packages/nocodb/src/controllers/attachments-secure.controller.ts

```diff
@@ -0,0 +1,71 @@
+import path from 'path';
+import {
+  Body,
+  Controller,
+  Get,
+  HttpCode,
+  Param,
+  Post,
+  Request,
+  Response,
+  UploadedFiles,
+  UseGuards,
+  UseInterceptors,
+} from '@nestjs/common';
+import hash from 'object-hash';
+import moment from 'moment';
+import { AnyFilesInterceptor } from '@nestjs/platform-express';
+import { GlobalGuard } from '~/guards/global/global.guard';
+import { AttachmentsService } from '~/services/attachments.service';
+import { TemporaryUrl } from '~/models';
+import { UploadAllowedInterceptor } from '~/interceptors/is-upload-allowed/is-upload-allowed.interceptor';
+
+@Controller()
+export class AttachmentsSecureController {
+  constructor(private readonly attachmentsService: AttachmentsService) {}
+
+  @UseGuards(GlobalGuard)
+  @Post(['/api/v1/db/storage/upload', '/api/v1/storage/upload'])
+  @HttpCode(200)
+  @UseInterceptors(UploadAllowedInterceptor, AnyFilesInterceptor())
+  async upload(@UploadedFiles() files: Array<any>, @Request() req) {
+    const path = `${moment().format('YYYY/MM/DD')}/${hash(req.user.id)}`;
```

Comment:
what would happen if I upload the same file multiple times / multiple files with the same name on the same day?

Reply:
we don't keep name it is a nanoid, so they will go into same folder but shouldn't clash